core/variant/array.cpp

```diff
@@ -202,9 +202,10 @@ Error Array::resize(int p_new_size) {
 	return _p->array.resize(p_new_size);
 }
 
-void Array::insert(int p_pos, const Variant &p_value) {
-	ERR_FAIL_COND(!_p->typed.validate(p_value, "insert"));
+Error Array::insert(int p_pos, const Variant &p_value) {
+	ERR_FAIL_COND_V(!_p->typed.validate(p_value, "insert"), ERR_METHOD_NOT_FOUND);
 	_p->array.insert(p_pos, p_value);
+	return OK;
```

Comment:
I'm not even sure why this ERR_FAIL is needed here.
Probably it will be needed to support typed arrays in future.
Error should be probably changed to `ERR_INVALID_PARAMETER` because it seems to validate if parameter is proper, not method.
Maybe some of this error will fit better:
https://github.com/godotengine/godot/blob/2abe996414b8b551e69e29461de3ff1bcaf5a28f/core/error/error_list.h#L41-L90

Reply:
Ah right, it is validating the arguement. Will change the error response.